azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -1385,6 +1385,9 @@ private void dispatch(final ExecutionReference reference, final ExecutableFlow e
     // move from flow to running flows
     this.runningFlows.put(exflow.getExecutionId(),
         new Pair<>(reference, exflow));
+    synchronized (this) {
```

Comment:
Could you add a comment to explain why notifyAll is needed? It may not be obvious to readers.

Reply:
Added that comment but it starts on the next line so this comment remains to be shown.